out_datadog: Document `dd_hostname` configuration option (#1401)
fluent/fluent-bit-docs: pipeline/outputs/datadog.md

| dd_message_key  | By default, the plugin searches for the key 'log' and remap the value to the key 'message'. If the property is set, the plugin will search the property name key.                                                                                                                                                                                                                                                                                                                                                                                                       |                                  |

> | dd_hostname     | The host that emitted logs should be associated with. If unset, Datadog will expect the host to be set as `host`, `hostname`, or `syslog.hostname` attributes. See Datadog Logs preprocessor documentation for up-to-date recognized attributes.                                                                                                                                                                                                                                                                                                                       |                                  |

Reviewer: Is there a link we can use to link the "Datadog Logs preprocessor documentation"?

Author: 👍 added a link in [c4d9263](https://github.com/fluent/fluent-bit-docs/pull/1401/commits/c4d9263d8836e599f7a186acf482b427de16a46a)

Reviewer: Nice! Thank you.

> | dd_message_key  | By default, the plugin searches for the key 'log' and remap the value to the key 'message'. If the property is set, the plugin will search the property name key.                                                                                                                                                                                                                                                                                                                                                                                                       |                                  |
> | dd_hostname     | The host that emitted logs should be associated with. If unset, Datadog will expect the host to be set as `host`, `hostname`, or `syslog.hostname` attributes. See Datadog Logs preprocessor documentation for up-to-date recognized attributes.                                                                                                                                                                                                                                                                                                                       |                                  |

Reviewer 2: The previously mentioned question is still applicable, but this suggestion updates the content to fit the new style guide, including having values be listed in ticks and not single quotes, and to not have empty cells in tables. It also makes a few wording changes for clarity (hopefully the content is still correct based on those).

```suggestion
| dd_message_key  | By default, the plugin searches for the key `log` and remaps the value to the key `message`. If the property is set, the plugin searches the property name key. | _none_ |
| dd_hostname     | The host the emitted logs should be associated with. If unset, Datadog expects the host to be set with `host`, `hostname`, or `syslog.hostname` attributes. See Datadog Logs preprocessor documentation for recognized attributes. | _none_ |
```

Author: Thanks @cnorris-cs ! The suggestion looks good to me. I applied it.
